dataRetriever/requestHandlers/sovereignRequestHandler.go

```diff
@@ -19,14 +25,95 @@ func NewSovereignResolverRequestHandler(resolverRequestHandler *resolverRequestH
 	return srrh, nil
 }
 
-// TODO: Implement this in MX-14496
-
 // RequestExtendedShardHeaderByNonce method asks for extended shard header from the connected peers by nonce
-func (srrh *sovereignResolverRequestHandler) RequestExtendedShardHeaderByNonce(_ uint64) {
-	//TODO: This method should be implemented for sovereign chain
+func (srrh *sovereignResolverRequestHandler) RequestExtendedShardHeaderByNonce(nonce uint64) {
+	suffix := fmt.Sprintf("%s_%d", uniqueHeadersSuffix, srrh.shardID)
+	key := []byte(fmt.Sprintf("%d-%d", srrh.shardID, nonce))
+	if !srrh.testIfRequestIsNeeded(key, suffix) {
+		return
+	}
+
+	log.Debug("sovereign: requesting extended shard header by nonce from network",
+		"shard", srrh.shardID,
+		"nonce", nonce,
+	)
+
+	requester, err := srrh.getExtendedShardHeaderRequester(srrh.shardID)
+	if err != nil {
+		log.Error("RequestShardHeaderByNonce.getExtendedShardHeaderRequester",
+			"error", err.Error(),
+			"shard", srrh.shardID,
+		)
+		return
+	}
+
+	headerRequester, ok := requester.(NonceRequester)
+	if !ok {
+		log.Warn("wrong assertion type when creating header requester")
+		return
+	}
+
+	srrh.whiteList.Add([][]byte{key})
+
+	epoch := srrh.getEpoch()
+	err = headerRequester.RequestDataFromNonce(nonce, epoch)
+	if err != nil {
+		log.Debug("RequestShardHeaderByNonce.RequestDataFromNonce",
```

Comment:
"RequestExtendedShardHeaderByNonce.RequestDataFromNonce"

Reply:
Changed it